### PR TITLE
JavaScript => TypeScript

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    "target": "es2017",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
+    // "outFile": "./",                       /* Concatenate and emit output to single file. */
+    // "outDir": "./",                        /* Redirect output structure to the directory. */
```
Comment (vladbarosan) [inline tsconfig.json]: maybe have the tsconfig in the root of the project ?

Comment (vladbarosan) [reply tsconfig.json]: nevermind, it seems there is a bug in codeflow showing these as in a folder

---
In reply to: [191033882](https://github.com/Azure/oav/pull/252#discussion_r191033882) [](ancestors = 191033882)

**Comment by vladbarosan** [comment]
LGTM but maybe lets wait until next week so we can test a little by pulling from your branch. otherwise its a long weekend and who knows who took a dependency on master directly :)

File: lib/autorestPlugin/pluginHost.ts [hunk 2]
```diff
@@ -19,21 +19,24 @@ const extension = new extensionBase.AutoRestExtension();
...
-  this.text = text;
-  this.source = source;
+class FormattedOutput{
+  constructor(
+    public channel: any,
+    public details: any,
+    public code: any,
+    public text: any,
+    public source: any,
+    _?: any) {
```
Comment (vladbarosan) [inline lib/autorestPlugin/pluginHost.ts]: what is this for ?

Comment (sergey-shandar) [reply lib/autorestPlugin/pluginHost.ts]: there's a call that uses more parameters than declared. It's definitely a bug that we need to fix. Ignore this for now :-)

Comment (vladbarosan) [reply lib/autorestPlugin/pluginHost.ts]: ok

**Pull request merged**